src/Polysemy/Internal/Union.hs

```diff
@@ -227,14 +228,19 @@ weaken (Union n a) = Union (SS n) a
 
 ------------------------------------------------------------------------------
 -- | Lift an effect @e@ into a 'Union' capable of holding it.
-inj :: forall r e a m. (Functor m , Member e r) => e m a -> Union r m a
-inj e = Union (finder @_ @r @e) $
+inj :: forall e r m a. (Functor m , Member e r) => e m a -> Union r m a
```

Comment:
Hmm, I rearranged this to fit the other functions in `Polysemy.Internal`, but under further consideration, perhaps this breaks something in `polysemy-plugin` or some of the TH machinery? I don't think this breaks anything inside TH, since I got everything to compile and the tests to complete without issue.

Reply:
and when I mean other functions, I apparently only mean `weaken`; `prj` has `e r a m` instead.

Reply:
I've now also rearranged the type vars in `prj` to fit the `e r m a` scheme, if you're ok with it.